app/controllers/deploys_controller.rb

```diff
@@ -178,4 +184,22 @@ def find_deploy
   def active_deploy_scope
     current_project ? current_project.deploys.active : Deploy.active
   end
+
+  # Creates a CSV for @deploys as a result of the search query limited to 1000 for speed
+  def as_csv
+    max = 1000
+    csv_limit = [(params[:limit].presence || max).to_i, max].min
+    # may be expensive to count an unfiltered @deploys so do count once
+    deploy_count = @deploys.limit(csv_limit + 1).count
```

Comment:
don't think you can limit a count ... also still doing 2 queries ... 

deploys = @deploys.limit(x+1).to_a
deploys[0..-2].each do |deploy|
if deploys.length > limit